crates/provider-sqldb-postgres/wit/deps/postgres/types.wit

```diff
@@ -0,0 +1,262 @@
+package wasmcloud:postgres@0.1.0-draft;
+
+/// Types used by components and providers of a SQLDB Postgres interface
+interface types {
+  /// This type of floating point is necessary as rust does not allow Eq/PartialEq/Hash on real `f64`
+  /// Instead we use a sign + mantissa + exponent
+  ///
+  /// see: https://docs.rs/num/latest/num/trait.Float.html#tymethod.integer_decode
+  type hashable-f64 = tuple<u64, s16, s8>;
+  type hashable-f32 = hashable-f64;
+
+  type point = tuple<hashable-f64, hashable-f64>;
+  type lower-left-point = point;
+  type upper-right-point = point;
+  type start-point = point;
+  type end-point = point;
+  type center-point = point;
+  type radius = hashable-f64;
+
+  type ipv4-addr = string;
+  type ipv6-addr = string;
+  type subnet = string;
+
+  type xmin = s64;
+  type xmax = s64;
+  type xip-list = list<s64>;
+
+  type logfile-num = u32;
+  type logfile-byte-offset = u32;
+
+  type column-name = string;
+
+  /// Arbitrary precision numeric type
+  type numeric = string;
+
+  /// Chosen weight of a Lexeme
+  enum lexeme-weight {
+    A,
+    B,
+    C,
+    D, // default
+  }
+
+  /// Represents an arbitrary precision numeric type
+  record lexeme {
+    /// Position (1->16383)
+    position: option<u16>,
+    /// Weight of the lexeme (in a relevant ts-vector)
+    weight: option<lexeme-weight>,
+    /// Data
+    data: string,
+  }
+
+  /// Offsets are expressed in seconds of timezone difference in either from the
+  /// eastern hemisphere or western hemisphere.
+  ///
+  /// ex. "America/New York", which is UTC-4 can be expressed as western-hemisphere-secs(4 * 3600)
+  variant offset {
+    eastern-hemisphere-secs(s32),
+    western-hemisphere-secs(s32),
+  }
+
+  /// Dates are represented similarly to tokio-postgres implementation
+  /// see: https://docs.rs/postgres-types/0.2.6/postgres_types/enum.Date.html#variant.Value
+  variant date {
```

Comment:
Question: Would it be better to use wasi-clocks' [datetime type](https://github.com/WebAssembly/wasi-clocks/blob/6cfb0f8fc33000f29652657efe144953846f80ee/wit/wall-clock.wit#L18) and then convert to postgres times from there?

Reply:
I don't think so -- there's quite a difference between this and that type, also this type is much easier to construct, and more accurate/reflecting of postgres's understanding of dates.